src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -162,127 +114,102 @@ public async Task<(Error, IEnumerable<GitHubUser>)> GetUsersByCommit(string owne
-                                                user = new
-                                                {
-                                                    databaseId = 0,
-                                                    login = "",
-                                                    name = "",
-                                                },
-                                            },
-                                        },
-                                    },
-                                },
-                            },
-                        },
-                    },
-                });
-
-                if (grahpApiResponse.errors != null && grahpApiResponse.errors.Length != 0)
-                {
-                    var notFoundError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "NOT_FOUND" && e.path.Contains("repository"));
-                    if (notFoundError != null)
-                    {
-                        // owner/repo doesn't exist or you don't have access to the repo
-                        _unknownRepos.TryAdd((owner, name));
+                            Id = node.author.user?.databaseId,
+                            Login = node.author.user?.login,
+                            Name = string.IsNullOrEmpty(node.author.user?.name) ? node.author.user?.login : node.author.user?.name,
+                            Emails = new[] { node.author.email, node.author.user?.email }
+                                .Where(email => !string.IsNullOrEmpty(email)).ToArray(),
+                        });
                     }
+                }
+            }
+
+            return (null, githubUsers);
+        }
+
+        private async Task<(Error error, T data, bool notFound)> Query<T>(string query, object variables, T dataType)
+        {
+            Debug.Assert(dataType != null);
 
-                    var rateLimitError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "MAX_NODE_LIMIT_EXCEEDED" || e.type == "RATE_LIMITED");
-                    if (rateLimitError != null)
+            var request = JsonUtility.Serialize(new { query, variables });
+
+            try
+            {
+                using (var response = await HttpPolicyExtensions
+                    .HandleTransientHttpError()
+                    .Or<OperationCanceledException>()
+                    .Or<IOException>()
+                    .RetryAsync(3)
+                    .ExecuteAsync(() => _httpClient.PostAsync(s_url, new StringContent(request, Encoding.UTF8, "application/json"))))
+                {
+                    if (response.StatusCode == HttpStatusCode.Unauthorized || response.StatusCode == HttpStatusCode.Forbidden)
                     {
-                        _rateLimitError = Errors.GitHubApiFailed(rateLimitError.message);
+                        Log.Write(await response.Content.ReadAsStringAsync());
+                        _fatalError = Errors.GitHubApiFailed(response.StatusCode.ToString());
```

Comment:
Is this message clear enough to expose to end user?

Reply:
Yes, there are more descriptions in the error message template.